force-app/main/default/classes/ClassesObjectsMethods.cls

```diff
@@ -163,126 +196,164 @@ public with sharing class ClassesObjectsMethods {
      */
     public static Integer generateRandomNumber() {
         // Use Math class to generate number between 50 and 100
-
+        Integer randomNumber = Integer.valueOf(Math.random());
+        Integer scaledRandomNumber = (Integer)(randomNumber * 51);
+        Integer finalRandomNumber = scaledRandomNumber + 50;
+        return finalRandomNumber;
```

Comment:
This will  work.  But I don't think you need all those variables. 

    return Math.round(Math.random() * 51) + 50;